### PR TITLE
adding requirements file for sdp delivery

### PR DESCRIPTION
**Pull request by nden**
This ended up more than I expected.
- Added dependency requirements for the upcoming release
- Fixed documentation build
- Added/removed documentation so that it reflects the state of the pipeline code.

### CODE REVIEW

**Review by ddavis-stsci — APPROVED** [review]
The only thing I see is that we still have "Keyword" in references_general.rst, but I don't think we need to change it now.

**Review by dmggh — APPROVED** [review]
Is error propagation for flat-fielding to be added later once the variance arrays of the SCI exposure exist ?

File: docs/roman/references_general/references_general.rst [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+| :ref:`FLAT <flat_reffile>`                       | :ref:`flatfield <flatfield_step>`           |
++--------------------------------------------------+---------------------------------------------+
+
+.. _`Standard ASDF metadata`:
+
+Standard ASDF metadata
+======================
+
+Al Roman science and reference files are ASDF files.
+
```
Comment (dmggh) [inline docs/roman/references_general/references_general.rst:63]: Is 'Keyword' (for asdf) equivalent to its FITS usage ?

**Comment by nden** [comment]
@dmggh Yes, Once we have the flat field step implemented we'll start adding the error propagation.

**Review by dmggh — COMMENTED** [review]
Is 'Keyword' (for asdf) equivalent to its FITS usage ?

File: docs/roman/references_general/references_general.rst (hunk 1 above)
Comment (nden) [reply docs/roman/references_general/references_general.rst:63]: `Keyword` is not generally used for ASDF but yes, it's the closest to FITS equivalent. 
In YAML the term is `field` which is not too popular. So after talking it over with others we agreed on using `attribute`

**Pull request merged**